lib/spack/spack/spec.py

```diff
@@ -2330,6 +2330,10 @@ def concretize(self, tests=False):
         # TODO: internal configuration conflicts within one package.
         matches = []
         for x in self.traverse():
+            if x.external:
+                # external specs are already built, don't worry about whether
+                # it's possible to build that configuration with Spack
+                continue
```

Comment:
I agree this is a shortcoming of the current concretizer, but it's not clear to me why we should be solving this here instead of suggesting a workaround in `packages.yaml`:
```yaml
packages:
  cmake:
    externals:
    - spec: cmake@3.18.1 %apple-clang
      prefix: /Users/culpo/PycharmProjects/spack/opt/spack/darwin-catalina-x86_64/apple-clang-11.0.0/cmake-3.18.1-rkhbrznb73yds25caqhypjowwlecgfne
    buildable: false
```
With the above I can do:
```console
% spack spec openjpeg %gcc            
Input spec
--------------------------------
openjpeg%gcc

Concretized
--------------------------------
openjpeg@2.3.1%gcc@9.2.0 build_type=RelWithDebInfo arch=darwin-catalina-x86_64
    ^cmake@3.18.1%apple-clang@11.0.0~doc+ncurses+openssl+ownlibs~qt arch=darwin-catalina-x86_64
    ^zlib@1.2.11%gcc@9.2.0+optimize+pic+shared arch=darwin-catalina-x86_64
```
Is there some other use case for which this change would be useful and that can't be treated like the above (or becomes very inconvenient)?

Reply:
CMake is easy, because we never really care what compiler it was built with. But you can imagine for another library a site may have hacked together a custom install of library `foo` built `%pgi`, even though `foo` doesn't typically build  `%pgi`. Users should still be able to properly refer to that external build in their packages.yaml file.

Reply:
I see where this is going, but I could argue that if `foo %pgi` is doable as an external hack then it might be replicated in our recipe (in which case it shouldn't be modeled as a conflict). In any case I don't see any harm in skipping conflicts check for externals, so I am going to give a green light on this.